lib/java/PlagiarismDetection/src/pd/SimComparer.java

```diff
@@ -195,144 +199,302 @@ private void gst(NGramList s1NGrams,
 		do {
 			maxMatch[0] = minMatch;
 			mappings = new ArrayList<Mapping>();
-			mappings = gstPhase1(s1Tokens, s1NGrams, s2NGramIndices, s2Tokens,
+			mappings = gstPhase1(s1Tokens, s1NGrams, null, s2Tokens,
 					maxMatch, mappings);
-			gstPhase2(mappings, s1Tokens, nGramSize, s1NGrams, s2Tokens,
-					bTokens, bNGramIndices, minMatch, tokenMappings);
+			gstPhase2(mappings, nGramSize, s1Tokens, s1NGrams, s2Tokens, s2NGrams,
+					bTokens, skeletonNGrams, minMatch, tokenMappings);
 
 		} while (maxMatch[0] > minMatch);
 	}
+	
+    private NGramList l1 = new NGramList(); // List of NGrams in s1 that are start of statements
+	private NGramList l2 = new NGramList(); // List of NGrams in s2 that are start of statements
+	private ArrayList<Boolean> l1_marks = new ArrayList<>(); // Keeps track of markings for list l1
+	private ArrayList<Boolean> l2_marks = new ArrayList<>(); // Keeps track of markings for list l2
+
+	// Iterates over l and check if all Ngrams between start and end are marked
+    private boolean all_is_unmarked(NGramList l, int start, int end) {
+		boolean ans = true;
+
+		if (l.equals(l1) ) {
```

Comment:
Thanks for the comments. Variables' names are important to help keep track of code flow. The code presented here is not friendly for reviewers to understand.